src/main/java/com/microsoft/aad/msal4j/AadInstanceDiscoveryProvider.java

```diff
@@ -46,6 +54,41 @@
         }
     }
 
+    static InstanceDiscoveryResponse parseInstanceDiscoveryMetadata(String instanceDiscoveryJson) {
+
+        InstanceDiscoveryResponse instanceDiscoveryResponse;
+        try {
+            instanceDiscoveryResponse = JsonHelper.convertJsonToObject(
+                    instanceDiscoveryJson,
+                    InstanceDiscoveryResponse.class);
+
+            validate(instanceDiscoveryResponse);
+
+        } catch(Exception ex){
+            throw new MsalClientException("Error parsing instance discovery response. Data must be" +
```

Comment:
needs more spaces...

Reply:
What needs more spaces? The error message?

Reply:
Yes, "Data must be_..."

Reply:
I see, will update